### ISSUE TITLE
Support aggregate expressions and GROUP BY in the Dart DSL

### ISSUE THREAD

**Issue by simolus3** [simolus3/moor#357]
We should add a `groupBy` method on `JoinedSelectStatement` and support the most common aggregate functions (count, sum, etc.) out of the box.

We should also have some api to _not_ select whole tables.

**Comment by simolus3** [comment]
@AabhasS (and others of course), can you give some examples for a query with `group by` that you want to write in Dart? I want to make sure they're straightforward to write with the new api.

When supporting `GROUP BY`, we likely need to support some mechanism to join a table without including it in the result set. I've thought about something like
```dart
final amountOfTodos = todos.id.count();

// go through popular categories and load how many items there are for them:
final query = (select(categories)
     .join([innerJoin(todos, todos.category.equalsExp(categories.id), includeResults: false])
   )
  ..addColumns([amountOfTodos])
  ..groupBy(category.id, having: amountOfTodos.isBiggerThanValue(5))
  ..orderBy([OrderingTerm.desc(amountOfTodos)]);

final results = await query.get();
for (final row in results) {
  final category = row.readTable(categories);
  final amountOfTodos = row.read(amountOfTodos);

  print('${category.name} has $amountOfTodos todo entries');
}
```

**Comment by simolus3** [comment]
Done - this feature will be part of the next moor release. You can preview the documentation with examples [here](https://develop--moor.netlify.com/docs/advanced-features/joins/#group-by). If you have any questions or feedback about the api, please let me know.

**Comment by AabhasS** [comment]
Thanks. I guess this will solve the purpose.

**Comment by AabhasS** [comment]
Thanks.
I just wanted to know why can't we have complete control over the columns
we select from one table or a joined table.
An example:  SELECT(farmers.id, count(assets.id))

**Comment by simolus3** [comment]
@AabhasS You can _add_ custom expressions to select statements since moor 2.3. With moor 2.4, you'll also be able to exclude the default table columns from a query.

So for `SELECT farmers.*, count(assets.id)`, you could use
```dart
final count = assets.id.count();
final query = (
  select(farmers)
    .join([innerJoin(assets, ..., useColumns: false)])
  )
  ..addColumns([count]);
```
I actually overlooked one aspect which means you can't remove columns from `farmers`. Good thing this wasn't released yet, I'll fix that :)

**Comment by AabhasS** [comment]
Yes, that is what I meant.

**Comment by simolus3** [comment]
It's now possible to call `join` on a `JoinedSelectStatement`, so this is possible:
```dart
final query = selectOnly(farmers).join([
  innerJoin(
    assets,
    assets.farmer.equalsExp(farmers.id),
    useColumns: false,
  )
]);
query
  ..addColumns([farmers.id, assets.id.count()])
  ..groupBy([farmers.id]);
```
This will only select `farmers.id` and `COUNT(assets.id)`.

**Comment by AabhasS** [comment]
Was this feature included in v2.1.0 that released on 1st Feb?
